src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -74,12 +80,112 @@ private void Initialize()
+                }
+            }
+        }
+
+        private void CacheResourceDeclaration(ResourceDeclarationSyntax resourceDeclarationSyntax, string type, string template)
+        {
+            if (!resourceTypeToBodyMap.ContainsKey(type))
+            {
+                TextSpan bodySpan = resourceDeclarationSyntax.Value.Span;
+                string bodyText = template.Substring(bodySpan.Position, bodySpan.Length);
+
+                resourceTypeToBodyMap.Add(type, bodyText);
+            }
+        }
+
+        private void CacheResourceDependencies(ImmutableHashSet<ResourceDependency> resourceDependencies, string template, string resourceType)
+        {
+            if (resourceDependencies.Any())
+            {
+                StringBuilder sb = new StringBuilder();
+                foreach (ResourceDependency resourceDependency in resourceDependencies)
+                {
+                    TextSpan span = resourceDependency.Resource.DeclaringSyntax.Span;
+                    sb.AppendLine(template.Substring(span.Position, span.Length));
+                }
+
+                resourceTypeToDependentsMap.Add(resourceType, sb.ToString());
+            }
+        }
+
+        private (ResourceDeclarationSyntax?, string?) GetResourceDeclarationSyntaxWithType(DeclaredSymbol declaredSymbol)
+        {
+            if (declaredSymbol.DeclaringSyntax is ResourceDeclarationSyntax resourceDeclarationSyntax &&
+                resourceDeclarationSyntax.TypeString is StringSyntax stringSyntax)
+            {
+                return (resourceDeclarationSyntax, stringSyntax.StringTokens.First().Text);
+            }
+
+            return (null, null);
+        }
+
+        private ImmutableDictionary<DeclaredSymbol, ImmutableHashSet<ResourceDependency>> GetResourceDependencies(string template, string manifestResourceName)
+        {
+            string path = Path.GetFullPath(manifestResourceName);
+            SyntaxTree syntaxTree = SyntaxTree.Create(new Uri(path), template);
+            SyntaxTreeGrouping syntaxTreeGrouping = new SyntaxTreeGrouping(
+                syntaxTree,
+                ImmutableHashSet.Create(syntaxTree),
+                ImmutableDictionary.Create<ModuleDeclarationSyntax, SyntaxTree>(),
+                ImmutableDictionary.Create<ModuleDeclarationSyntax, DiagnosticBuilder.ErrorBuilderDelegate>());
+
+            Compilation compilation = new Compilation(new AzResourceTypeProvider(), syntaxTreeGrouping);
+            SemanticModel semanticModel = compilation.GetEntrypointSemanticModel();
+
+            return ResourceDependencyVisitor.GetResourceDependencies(semanticModel);
+        }
+
+        public Snippet? GetResourceBodyCompletionSnippet(string type)
+        {
+            if (string.IsNullOrWhiteSpace(type))
+            {
+                return null;
+            }
+
+            Snippet? snippet = topLevelNamedDeclarationSnippets.Where(x => x.Text.Contains(type)).FirstOrDefault();
```

Comment:
>Contains [](start = 82, length = 8)

Could this match incorrectly if the snippet contains a comment or other text that belongs to a different type?

Reply:
I updated it here to keep the label and description - {} for snippets sourced from static template file and swagger. Let me know if there are any concerns. 

https://github.com/Azure/bicep/commit/9b3995e29ee139dea8322b474ded7a2b4a34cc94

Reply:
Marking this resolved. Let me know if you have any questions.